Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm

```diff
@@ -99,10 +99,15 @@ static bool determineTrackingPreventionStateInternal(bool appWasLinkedOnOrAfter,
     ASSERT(!RunLoop::isMain());
     ASSERT(!isInWebKitChildProcess());
 
-    if (!appWasLinkedOnOrAfter && !isFullWebBrowser(bundleIdentifier))
+#if ENABLE(APP_BOUND_DOMAINS)
+    bool isFullWebBrowser = isFullWebBrowserOrRunningTest(bundleIdentifier);
+#else
+    bool isFullWebBrowser = isRunningTest(bundleIdentifier);
+#endif
+    if (!appWasLinkedOnOrAfter && !isFullWebBrowser)
         return false;
 
-    if (!isFullWebBrowser(bundleIdentifier) && !hasRequestedCrossWebsiteTrackingPermission())
+    if (!isFullWebBrowser && !hasRequestedCrossWebsiteTrackingPermission())
```

Comment:
isFullWebBrowser is a bit misleading, it would be more something like isTrackingPreventionEntitledApplication, which is a bit long.